Runtime/CesiumTextureUtility.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+
+namespace CesiumForUnity
+{
+    public sealed class CesiumTextureUtility : MonoBehaviour
+    {
+        private CesiumTextureUtility() { }
+
+        private static CesiumTextureUtility _instance = null;
+
+        public static CesiumTextureUtility Instance
+        {
+            get
+            {
+                if (_instance == null) _instance = new CesiumTextureUtility();
+                return _instance;
+            }
+        }
+
+        private bool _initialized = false;
+
+        public bool ETC1_RGB { get; set; } = false;
+
+        public bool ETC2_RGBA { get; set; } = false;
+
+        public bool BC1_RGB { get; set; } = false;
+
+        public bool BC3_RGBA { get; set; } = false;
+
+        public bool BC4_R { get; set; } = false;
+
+        public bool BC5_RG { get; set; } = false;
+
+        public bool BC7_RGBA { get; set; } = false;
+
+        public bool PVRTC1_4_RGB { get; set; } = false;
+
+        public bool PVRTC1_4_RGBA { get; set; } = false;
+
+        public bool ASTC_4x4_RGBA { get; set; } = false;
+
+        public bool PVRTC2_4_RGB { get; set; } = false;
+
+        public bool PVRTC2_4_RGBA { get; set; } = false;
+
+        public bool ETC2_EAC_R11 { get; set; } = false;
+
+        public bool ETC2_EAC_RG11 { get; set; } = false;
+
+        public void CheckSupportedGpuCompressedPixelFormats()
+        {
+            if (_initialized)
+                return;
+
+            ETC1_RGB = SystemInfo.IsFormatSupported(GraphicsFormat.RGB_ETC_UNorm, FormatUsage.Sample);
+            ETC2_RGBA = SystemInfo.IsFormatSupported(GraphicsFormat.RGBA_ETC2_SRGB, FormatUsage.Sample);
+            BC1_RGB = SystemInfo.IsFormatSupported(GraphicsFormat.RGBA_DXT1_SRGB, FormatUsage.Sample);
+            BC3_RGBA = SystemInfo.IsFormatSupported(GraphicsFormat.RGBA_DXT5_SRGB, FormatUsage.Sample);
+            BC4_R = SystemInfo.IsFormatSupported(GraphicsFormat.R_BC4_SNorm, FormatUsage.Sample);
+            BC5_RG = SystemInfo.IsFormatSupported(GraphicsFormat.RG_BC5_SNorm, FormatUsage.Sample);
+            BC7_RGBA = SystemInfo.IsFormatSupported(GraphicsFormat.RGBA_BC7_SRGB, FormatUsage.Sample);
+            ASTC_4x4_RGBA = SystemInfo.IsFormatSupported(GraphicsFormat.RGBA_ASTC4X4_SRGB, FormatUsage.Sample);
+
+            //not ready in TextureLoader::loadTexture() (path: native~/Runtime/src/TextureLoader.cpp)
```

Comment:
What would it take to get them ready? They're supported in Cesium for Unreal I believe.

Reply:
I didn't assign value of textureFormat in these cases in native~/Runtime/src/TextureLoader.cpp

-  case GpuCompressedPixelFormat::PVRTC1_4_RGB:
-  case GpuCompressedPixelFormat::PVRTC1_4_RGBA:
-  case GpuCompressedPixelFormat::PVRTC2_4_RGB:
-  case GpuCompressedPixelFormat::PVRTC2_4_RGBA:
-  case GpuCompressedPixelFormat::ETC2_EAC_R11:
-  case GpuCompressedPixelFormat::ETC2_EAC_RG11:

for PVRTC1 I don't have a device (IOS or something PVRTC1 supported device) so I didn't test it and I am not cofident with PVRTC1

and I coudn't find matched type for these in UnityEngine.Experimental.Rendering.GraphicsFormat enum.

- PVRTC2_4_RGB
- PVRTC2_4_RGBA
- ETC2_EAC_R11
- ETC2_EAC_RG11

Reply:
I didn't assign value of textureFormat in these cases in native~/Runtime/src/TextureLoader.cpp

- case GpuCompressedPixelFormat::PVRTC1_4_RGB:
- case GpuCompressedPixelFormat::PVRTC1_4_RGBA:
- case GpuCompressedPixelFormat::PVRTC2_4_RGB:
- case GpuCompressedPixelFormat::PVRTC2_4_RGBA:
- case GpuCompressedPixelFormat::ETC2_EAC_R11:
- case GpuCompressedPixelFormat::ETC2_EAC_RG11:

for PVRTC1 I don't have a device (IOS or something PVRTC1 supported device) so I didn't test it and I am not cofident with PVRTC1

and I coudn't find matched type for these in UnityEngine.Experimental.Rendering.GraphicsFormat enum.

- PVRTC2_4_RGB
- PVRTC2_4_RGBA
- ETC2_EAC_R11
- ETC2_EAC_RG11